src/timer.c

```diff
@@ -129,8 +214,9 @@ qd_timestamp_t qd_timer_now() {
 void qd_timer_schedule(qd_timer_t *timer, qd_duration_t duration)
 {
     sys_mutex_lock(lock);
-    timer_cancel_LH(timer);  // Timer is now on the idle list
-    DEQ_REMOVE(idle_timers, timer);
+
```

Comment:
If timer state is indeed QD_TIMER_STATE_DELETED should we return immediately ?

Reply:
If state is DELETED that indicates a bug where a codepath is attempting to schedule a timer that has been deleted.  Returning immediately would mask that bug - better to assert().